lib/rubygems/package/tar_reader/entry.rb

```diff
@@ -157,13 +187,62 @@ def readpartial(maxlen = nil, outbuf = "".b)
     outbuf
   end
 
+  ##
+  # Seeks to +offset+ bytes into the tar file entry
+  # +whence+ can be IO::SEEK_SET, IO::SEEK_CUR, or IO::SEEK_END
+
+  def seek(offset, whence = IO::SEEK_SET)
+    check_closed
+
+    new_pos =
+      case whence
+      when IO::SEEK_SET then @orig_pos + offset
+      when IO::SEEK_CUR then @io.pos + offset
+      when IO::SEEK_END then @orig_pos + @header.size + offset
```

Comment:
Maybe I would extract `@end_pos = @orig_pos + @header.size` in `initialize`, just to bring nice symmetry to this logic.

Reply:
nice, I'll do that.